### PR TITLE
[Bugfix] Avoid get SQL request headers when SET search_path is added

### LINKED ISSUE #287: Erreur SQL lors de la consultation des infos parcelle
**Description du bug**

Bonjour,
Impossible de consulter les relevés de propriétés, les informations parcelles, etc. Les onglets de la boite d'information restent vides exception faite des titres (Parcelle,...)
Mes collègues ne sont pas sur la même version de QGis que moi (les leurs sont plus anciennes) et ne sont pas affectés.

Je soupçonne le SET search_path = "cadastre", public, pg_catalog; de la requête de provoquer ces erreurs, puisque lorsque je les supprime, j'ai bien des résultats dans ma base.

**Reproduire le bug**

Import des données : ok.
Outil consultation info parcellaire

**Log**

Aucun log d'erreur du plugin
`
Log Erreur Postgis :
(début)

```
2021-03-16T16:42:24     WARNING    Requête erronée : SELECT * FROM (SELECT row_number() OVER () AS __rid__, * FROM (SET search_path = "cadastre", public, pg_catalog;SELECT
             
             '
              '
Code
' || ccosec || dnupla || '
' ||
              '
Commune
' || c.libcom || '
' ||
              '
Date de l''acte
' || COALESCE(Cast(p.jdatat AS text), '') || '
' ||
              '
Surface cadastrale (contenance)
' || p.dcntpa || ' m²
' ||
              '
Surface géographique
' || round(ST_Area(gp.geom)) || ' m²
' ||
              '
Surface bâtie
' || coalesce(sum(round(ST_Area(b.geom))), 0) || ' m²
' ||
              '
Pourcentage du bâti
' || coalesce(round(100 * sum(ST_Area(b.geom)) / ST_Area(gp.geom)), 0) || '
' ||
             
              '
Adresse
' ||
              coalesce(
              CASE
              WHEN v.libvoi IS NOT NULL THEN trim(ltrim(p.dnvoiri, '0') || ' ' || trim(coalesce(v.natvoi, '')) || ' ' || v.libvoi)
              ELSE ltrim(p.cconvo, '0') || p.dvoilib
              END, '') ||
              '
' ||
              '
Urbaine
' ||
              CASE
              WHEN p.gurbpa = 'U' THEN 'Oui'
              ELSE 'Non'
              END ||
              '
' ||
             '
'
             
             FROM parcelle p
             INNER JOIN geo_parcelle gp ON p.parcelle = gp.geo_parcelle
             LEFT OUTER JOIN geo_batiment b
              ON ST_Intersects(ST_Centroid(b.geom), gp.geom)
             LEFT OUTER JOIN commune c
              ON p.ccocom = c.ccocom AND c.ccodep = p.ccodep
             LEFT OUTER JOIN voie v
              ON v.voie = p.voie
             WHERE 2>1
             AND parcelle = '510454000AT0068'
             GROUP BY p.ccosec, p.dnupla, c.libcom, p.jdatat, p.dcntpa, gp.geom, v.libvoi, p.dnvoiri, v.natvoi, p.cconvo, p.dvoilib, p.gurbpa
             LIMIT 1
              LIMIT 1 ) as foo) AS "subQuery_0" LIMIT 1 a retourné 7 [ERREUR: erreur de syntaxe sur ou près de « = »
             LINE 1: ...ber() OVER () AS __rid__, * FROM (SET search_path = "cadastr...
              ^
             ]
2021-03-16T16:42:24     WARNING    Échec de l'exécution de la requête.
             Le message d'erreur de la base de données est :
             ERREUR: erreur de syntaxe sur ou près de « = »
             LINE 1: ...ber() OVER () AS __rid__, * FROM (SET search_path = "cadastr...
              ^
             .
             SQL : SELECT * FROM (SELECT row_number() OVER () AS __rid__, * FROM (SET search_path = "cadastre", public, pg_catalog;SELECT
             
             '
              '
Code
' || ccosec || dnupla || '
' ||
              '
Commune
' || c.libcom || '
' ||
              '
Date de l''acte
' || COALESCE(Cast(p.jdatat AS text), '') || '
' ||
              '
Surface cadastrale (contenance)
' || p.dcntpa || ' m²
' ||
              '
Surface géographique
' || round(ST_Area(gp.geom)) || ' m²
' ||
              '
Surface bâtie
' || coalesce(sum(round(ST_Area(b.geom))), 0) || ' m²
' ||
              '
Pourcentage du bâti
' || coalesce(round(100 * sum(ST_Area(b.geom)) / ST_Area(gp.geom)), 0) || '
' ||
             
              '
Adresse
' ||
              coalesce(
              CASE
              WHEN v.libvoi IS NOT NULL THEN trim(ltrim(p.dnvoiri, '0') || ' ' || trim(coalesce(v.natvoi, '')) || ' ' || v.libvoi)
              ELSE ltrim(p.cconvo, '0') || p.dvoilib
              END, '') ||
              '
' ||
              '
Urbaine
' ||
              CASE
              WHEN p.gurbpa = 'U' THEN 'Oui'
              ELSE 'Non'
              END ||
              '
' ||
             '
'
             
             FROM parcelle p
             INNER JOIN geo_parcelle gp ON p.parcelle = gp.geo_parcelle
             LEFT OUTER JOIN geo_batiment b
              ON ST_Intersects(ST_Centroid(b.geom), gp.geom)
             LEFT OUTER JOIN commune c
              ON p.ccocom = c.ccocom AND c.ccodep = p.ccodep
             LEFT OUTER JOIN voie v
              ON v.voie = p.voie
             WHERE 2>1
             AND parcelle = '510454000AT0068'
             GROUP BY p.ccosec, p.dnupla, c.libcom, p.jdatat, p.dcntpa, gp.geom, v.libvoi, p.dnvoiri, v.natvoi, p.cconvo, p.dvoilib, p.gurbpa
             LIMIT 1
              LIMIT 1 ) as foo) AS "subQuery_0" LIMIT 1
```

**Environnement**
 - OS: Windows10
 - Version de QGIS :  3.16.0-Hannover 
 - Version du plugin : Version 1.10.2 
 - PostgreSQL : 9.6

### PR DESCRIPTION
**Pull request by Gustry**
For some request, the `SET search_path` is added to the SQL. This is not compatible with DBPostGISConnector description that using QgsVectorLayer to get headers.

Fixes #287

Superseded #292 

@rldhont Thanks for your previous PR #292, it really helps. This PR is just to avoid duplicated code to maintain. I prefer having a new parameter than copy/paste a block of code. What do you think ?
The code of the plugin with the usage of DBManager is already difficult to maintain.

I haven't try the PR. I just took it from the previous PR.

### CODE REVIEW

File: cadastre/cadastre_common_base.py [hunk 1]
```diff
@@ -389,7 +391,7 @@ def getProprietaireComptesCommunaux(comptecommunal: str, connectionParams: Dict[
     if connectionParams['dbType'] == 'spatialite':
         sql = sql.replace('MyStringAgg', 'group_concat')
 
-    [header, data, rowCount, ok] = fetchDataFromSqlQuery(connector, sql)
+    _, data, _, ok = fetchDataFromSqlQuery(connector, sql, with_header=False)
     ccs = []
     if ok:
         for line in data:
```
Comment (rldhont) [inline cadastre/cadastre_common_base.py:397]: I have added a `break` here

**Comment by rldhont** [comment]
Good to me even if I prefer simpliest API, you can close #292.

As discuss with @mdouchin we have to check if headers are used `fetchDataFromSqlQuery`

**Comment by Gustry** [comment]
> Good to me even if I prefer simpliest API, you can close #292.

Je comprends, mais comme on a du mal à maintenir ces milliers de lignes de code, je cherche aussi à ne pas dupliquer du code, surtout qu'il est voué à être supprimer si on l'utilise l'API QGIS pour interroger une base de données.

> As discuss with @mdouchin we have to check if headers are used fetchDataFromSqlQuery

Le retour de la fonction `fetchDataFromSqlQuery` avec `header` ?

Tu veux ca comme résultat ? :)

```bash
^C(.venv) etienne@longitude:~/dev/python/QgisCadastrePlugin (master)$ pylint cadastre | grep header
cadastre/cadastre_export.py:331:16: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_dialogs.py:276:16: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_dialogs.py:332:8: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_dialogs.py:1261:16: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_dialogs.py:1555:8: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_dialogs.py:1665:8: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_dialogs.py:2694:8: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_import.py:72:16: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_common_base.py:365:4: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_common_base.py:392:4: W0612: Unused variable 'header' (unused-variable)
cadastre/cadastre_common_base.py:451:4: W0612: Unused variable 'header' (unused-variable)
```

**Comment by Gustry** [comment]
> As discuss with @mdouchin we have to check if headers are used fetchDataFromSqlQuery

New PR #294

**Pull request closed without merging**